Android/src/org/droidplanner/android/proxy/mission/item/fragments/MissionDetailFragment.java

```diff
@@ -206,11 +208,69 @@ public void onSpinnerItemSelected(Spinner spinner, int position) {
         }
     };
 
+    private final Spinner.OnItemSelectedListener missionFrameListener = new Spinner.OnItemSelectedListener() {
+
+        @Override
+        public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+            frameSpinner = (Spinner) view.findViewById(R.id.frameSpinner);
+
+            if (frameSpinner != null && mSelectedProxies.isEmpty())
+                return;
+
+            MissionItemType selectedType = commandAdapter.getItem(position);
+            List<Pair<MissionItemProxy, List<MissionItemProxy>>> updatedList = new ArrayList<>(
+                    mSelectedProxies.size());
+
+            for (MissionItemProxy missionItemProxy : mSelectedProxies) {
+
+                MissionItem currentItem = missionItemProxy.getMissionItem();
+
+                List<MissionItemProxy> updatedItems = new ArrayList<>();
```

Comment:
@billbonney Looks like there's at most one entry in the `updatedItems` list for every loop, hence the list doesn't seem needed.